src/MLDatasets.jl

```diff
@@ -2,28 +2,35 @@ module MLDatasets
 
 using FixedPointNumbers
 using SparseArrays
-using DataFrames, Tables
-using Glob
-import ImageCore
-using ColorTypes
+using Tables
+using DataDeps
 import MLUtils
 using MLUtils: getobs, numobs, AbstractDataContainer
-
-### I/O imports
-import NPZ
-import Pickle
-using MAT: matopen, matread
-import CSV 
-using HDF5
-using JLD2
-import JSON3
+using ColorTypes
+using Glob
 using DelimitedFiles: readdlm
-##########
+using FileIO
+using LazyModules: @lazy
 
-export getobs, numobs
-export convert2image
+include("require.jl") # export @require
 
+# Use `@lazy import SomePkg` whenever the returned types are not its own types,
+# since for methods applied on the returned types we would encounter in world-age issues
+# (see discussion in  https://github.com/JuliaML/MLDatasets.jl/pull/128).
+# In the other case instead, use `require import SomePkg` to force 
+# the use to manually import.
 
+@require import JSON3="0f8b85d8-7281-11e9-16c2-39a750bddbf1"
+@require import DataFrames="a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
```

Comment:
It seems that this package is loaded by some dependencies already:

```julia
julia> using MLDatasets

julia> dataset = Titanic()
dataset Titanic:
  metadata    =>    Dict{String, Any} with 5 entries
  features    =>    891×11 DataFrame
  targets     =>    891×1 DataFrame
  dataframe   =>    891×12 DataFrame
```

I would expect this to error for the first call but it isn't.

Reply:
for me it is erroring. Are you in the right branch?

Reply:
Oh my mistake; I forgot to activate the current folder 🙈